webapp/app/js/compare.js

```diff
@@ -0,0 +1,448 @@
+                           $scope.newProject + '/performance-data/0/' +
+                           'get_performance_data/?interval_seconds=' + $scope.timeRange;
+
+            //ok, now get the new revision
+            var signatureListURL = thServiceDomain + '/api/project/' + $scope.newProject + 
+              '/performance-data/0/get_performance_series_summary/?interval=' +
+              $scope.timeRange;
+
+            // TODO: figure how how to reduce these maps
+            var new_rawResultsMap = {};
+            var new_seriesList = [];
+
+
+            $http.get(signatureListURL).then(function(response) {
+              Object.keys(response.data).forEach(function(signature) {
+                var seriesSummary = getSeriesSummary(signature,
+                                                     response.data[signature],
+                                                     optionCollectionMap,
+                                                     $stateParams.pgo,
+                                                     $stateParams.e10s);
+
+                if (seriesSummary != null && seriesSummary.signature !== undefined) {
+                  new_seriesList.push(seriesSummary);
+                }
+              });
+
+
+              $q.all(new_seriesList.map(function(series) {
+                return $http.get(signatureURL + "&signatures=" + series.signature).then(function(response) {
+                  response.data.forEach(function(data) {
+                    new_rawResultsMap[data.series_signature] = calculateStats(data.blob, $scope.newResultSetID);
+                  });
+                });
+              })).then(function () {
+                displayResults(seriesList, rawResultsMap, new_seriesList, new_rawResultsMap);
+              });
+            });
+          });
+        }
+      );
+    }
+
+    function calculateStats(perfData, resultSetID) {
+      var geomeans = [];
+      var total = 0;
+      perfData.forEach(function(pdata) {
+        if (pdata.result_set_id != resultSetID) {
+          return null;
+        }
+
+        geomeans.push(pdata.geomean);
+        total += pdata.geomean;
+      });
+
+      var avg = total / geomeans.length;
+      var sigma = stddev(geomeans, avg);
+      return {'geomean': avg, 'variation': 2*sigma, 'runs': geomeans.length};
+    }
+
+    function displayResults(seriesList, rawResultsMap, new_seriesList, new_rawResultsMap) {
+      var counter = 0;
+      var compareResultsMap = {};
+
+      $scope.testList.forEach(function(testName) {
+        if (counter > 0 && compareResultsMap[(counter-1)].isHeader) {
+          counter--;
+        }
+
+        //TODO: figure out a cleaner method for making the names a header row
+        compareResultsMap[counter++] = {'name': testName, 'isHeader': true};
+
+        //TODO: need to figure out how to iterate through the old and new series list
+        //      Ideally a seriesList.forEach(...)
+        for (var idx in seriesList) {
```

Comment:
Actually wait, never mind, I see why you did that. Maybe there's something in underscore.js that will fit the bill though...